### ISSUE TITLE
Rubinius + examples/websockets.rb leads to "ArgumentError: Data object has already been freed" on master intermittently

### ISSUE THREAD

**Issue by justindarby** [celluloid/reel#136]
Just did a fresh clone and loaded up examples/websockets.rb to do some testing, keep getting "ArgumentError: Data object has already been freed" during my tests (and a reset client connection) when reusing connections.. only setup the example to reproduce another issue with a broken websocket client, but this is occuring during a GET /...

This is probably broken, though I don't have time right now to chase down who is causing that.

Full bt from console:
```
E, [2014-02-11T18:36:31.359193 #26780] ERROR -- : WebServer crashed!
ArgumentError: Data object has already been freed
        /usr/local/rubinius/gems/gems/reel-0.5.0.pre/lib/reel/request/parser.rb:46:in `readpartial'
        /usr/local/rubinius/gems/gems/reel-0.5.0.pre/lib/reel/request/parser.rb:39:in `current_request'
        /usr/local/rubinius/gems/gems/reel-0.5.0.pre/lib/reel/connection.rb:57:in `request'
        ./ReproTastic.rb:54:in `on_connection'
        kernel/common/method.rb:71:in `call'
        kernel/bootstrap/proc.rb:18:in `call'
        /usr/local/rubinius/gems/gems/reel-0.5.0.pre/lib/reel/server.rb:56:in `handle_connection'
        /usr/local/rubinius/gems/gems/celluloid-0.15.2/lib/celluloid/calls.rb:25:in `dispatch'
        /usr/local/rubinius/gems/gems/celluloid-0.15.2/lib/celluloid/calls.rb:122:in `dispatch'
        /usr/local/rubinius/gems/gems/celluloid-0.15.2/lib/celluloid/actor.rb:322:in `handle_message'
        /usr/local/rubinius/gems/gems/celluloid-0.15.2/lib/celluloid/actor.rb:416:in `task'
        /usr/local/rubinius/gems/gems/celluloid-0.15.2/lib/celluloid/tasks.rb:54:in `initialize'
        /usr/local/rubinius/gems/gems/celluloid-0.15.2/lib/celluloid/tasks/task_fiber.rb:13:in `create'
        kernel/bootstrap/proc.rb:20:in `call'
```

Let me know if you need more details.

**Comment by digitalextremist** [comment]
Issue is no longer present in jruby 1.7.16.1+

**Comment by digitalextremist** [comment]
Released in 0.6.0.pre1
